stix2/base.py

```diff
@@ -109,6 +102,13 @@ def __init__(self, allow_custom=False, **kwargs):
             extra_kwargs = list(set(kwargs) - set(cls._properties))
             if extra_kwargs:
                 raise ExtraPropertiesError(cls, extra_kwargs)
+        else:
+            from .properties import CustomProperty
+
+            # The custom properties will get added to the bottom.
+            # Matched with a CustomProperty.
+            extra_kwargs = list(set(kwargs) - set(cls._properties))
+            self._properties.update([(x, CustomProperty()) for x in extra_kwargs])
```

Comment:
I'm not sure we want to do this. It updates the class instance of `_properties`, meaning that any custom properties on the first instantiation of an SDO are allowed on any future instance, even if allow_custom is false for the future ones.

It's perhaps likely that people want to use the same custom properties on every instance of the same SDO that they create, but I'm not sure how I feel about the way this is done.

Reply:
That is a good point! So, could we (possible approaches):
- Make _properties an instance variable?
- Handle custom in a completely different structure?
- Go back with the old method (required checking the _inner dict)?

Reply:
I'll go back to the #85 fix. That was working fine.